packages/react/src/components/Forms/Switch/index.spec.tsx

```diff
@@ -0,0 +1,127 @@
+import { render, screen } from "@testing-library/react";
+import React from "react";
+import userEvent from "@testing-library/user-event";
+import { Switch } from ":/components/Forms/Switch/index";
+
+describe("<Switch/>", () => {
+  it("renders and can be checked", async () => {
+    const user = userEvent.setup();
+    render(<Switch label="Newsletter" />);
+    const input: HTMLInputElement = screen.getByRole("checkbox", {
+      name: "Newsletter",
+    });
+    expect(input.checked).toEqual(false);
+    await user.click(input);
+    expect(input.checked).toEqual(true);
+  });
+  it("renders with default value and can be unchecked", async () => {
+    const user = userEvent.setup();
+    render(<Switch label="Newsletter" defaultChecked={true} />);
+    const input: HTMLInputElement = screen.getByRole("checkbox", {
+      name: "Newsletter",
+    });
+    expect(input.checked).toEqual(true);
+    await user.click(input);
+    expect(input.checked).toEqual(false);
+  });
+  it("renders disabled", async () => {
+    render(<Switch label="Newsletter" disabled={true} />);
+    expect(screen.getByRole("checkbox", { name: "Newsletter" })).toBeDisabled();
+    // Click and expect the checkbox does not get checked
+    const user = userEvent.setup();
+    const input: HTMLInputElement = screen.getByRole("checkbox", {
+      name: "Newsletter",
+    });
+    expect(input.checked).toEqual(false);
+    await user.click(input);
+    expect(input.checked).toEqual(false);
+  });
+  it("renders with text", async () => {
+    render(<Switch label="Newsletter" text="Text" />);
+    expect(screen.getByText("Text")).toBeInTheDocument();
+  });
+  it("renders with state=success", async () => {
+    render(<Switch label="Newsletter" state="success" text="Success text" />);
+    expect(screen.getByText("Success text")).toBeInTheDocument();
+    expect(
+      document.querySelector(".c__field.c__field--success")
+    ).toBeInTheDocument();
+  });
+  it("renders with state=error", async () => {
+    render(<Switch label="Newsletter" state="error" text="Error text" />);
+    expect(screen.getByText("Error text")).toBeInTheDocument();
+    expect(
+      document.querySelector(".c__field.c__field--error")
+    ).toBeInTheDocument();
+  });
+  it("renders multiple", async () => {
+    // make sure switching one does not switch the others.
+    const user = userEvent.setup();
+    render(
+      <div>
+        <Switch label="Newsletter" />
+        <Switch label="Notifications" />
+        <Switch label="Phone" />
+      </div>
+    );
+    // expect all checkboxes to be unchecked
+    const newsletter: HTMLInputElement = screen.getByRole("checkbox", {
```

Comment:
you either use .toBeInTheDocument() with a getByX, and at other times you don't. Why ?